modules/ding_provider/connie/connie.loan.inc

```diff
@@ -45,3 +47,21 @@ function connie_loan_renew($account, $ids) {
 
   return $result;
 }
+
+/**
+ * Enrich pseudo entities.
+ *
+ * @param object $ting_entity
+ *   Provider reservation object.
+ * @param \Ting\TingObject $ting_object
+ *   Mutable ting object that represents the raw search result from the search
+ *   provider.
+ *
+ * @return object
+ *   Ting entity object, filled with data from provider service.
+ */
+function connie_loan_populate_pseudo_entity($ting_entity, TingObject $ting_object) {
+  $ting_object->setTitle(trim($ting_object->getTitle() . " Connie edition"));
+
+  return $ting_entity;
```

Comment:
This modifies `$ting_object` but returns `$ting_entity`. It is not clear how/if the entity has been modified. Perhaps it is an oddity based on the class contracts where the ting entity cannot return a mutable object 🤔 .

Reply:
yeah, that's pretty much it. I could actually just mutate $ting_entity and return nothing, but as the ding_provider.module assigns the return-value to its copy of $ting_entity that would end up overwriting the entity.

The alternative would be to just forego with returning anything and relying on pass-by reference.

... I'll give it a quick try.